### DISCUSSION TITLE
예제로 주문 넣어봤는데 주문 처리가 안됩니당

### DISCUSSION THREAD

**Discussion by llIllIllllIIIlllllll** [elbakramer/koapy#89] [Q&A] [answered]
koapy/koapy/examples/12_grpc_client_auth.py

koapy/koapy/examples/12_grpc_server_auth.py 

위예제 고대로 복사해서 
grpc_client_auth.py 

main 안에

로그인, 조건식, 계좌 정보는 잘 나오는데

```python
request_name = "삼성전자 1주 시장가 신규 매수"  # 사용자 구분명, 구분가능한 임의의 문자열
screen_no = "0001"  # 화면번호, 0000 을 제외한 4자리 숫자 임의로 지정, None 의 경우 내부적으로 화면번호 자동할당
account_no = "8014526011"  # 계좌번호 10자리, 여기서는 계좌번호 목록에서 첫번째로 발견한 계좌번호로 매수처리
order_type = 1  # 주문유형, 1:신규매수
code = "005930"  # 종목코드, 앞의 삼성전자 종목코드
quantity = 1  # 주문수량, 1주 매수
price = 0  # 주문가격, 시장가 매수는 가격 설정 의미 없으므로 기본값 0 으로 설정
quote_type = "03"  # 거래구분, 03:시장가
original_order_no = ""  # 원주문번호, 주문 정정/취소 등에서 사용

stream = entrypoint.OrderCall(request_name, screen_no, account_no, order_type, code, quantity, price, quote_type, original_order_no)

for event in warn_on_rpc_error(stream):
     if event.name == "OnReceiveTrData":
         order_no = event.single_data.values[0]
     elif event.name == "OnReceiveChejanData":
         gubun = event.arguments[0].string_value
         data = dict(zip(event.single_data.names, event.single_data.values))
         if gubun == "0":
             status = data["주문상태"]
             if status == "접수":
                 pass
             elif status == "체결":
                 orders_filled = data["체결량"]
                 orders_left = data["미체결수량"]
             elif status == "확인":
                 org_order_no = data["원주문번호"]
                 assert original_order_no == org_order_no
         elif gubun in ["1", "4"]:
             stocks = data["보유수량"]
```

그대로 넣었는데 반응이 없어요 오류도 안뜹니다...ㅠ

**Comment by elbakramer** [comment, marked as answer]
말씀주신 반응이 없다는 걸 좀 더 구체적으로 설명주세요.
예상했던 결과 VS 실제얻은 결과가 각각 어떠했고 어디서 달랐는지 구체적으로 말씀주시면 현상 파악에 도움 될 거 같습니다.

1. 말씀주신 반응이 단순 출력이 없다는 의미라면, 첨부주신 코드상 별도 출력코드가 없기 때문에 출력이 안되는게 자연스러운거 같습니다.
2. 반환된 이벤트 스트림에서 아무 이벤트도 발생하지 않는거라면 단순 오류일수도 있고, 아니면 장이 열린 시간이 아니라 주문이 안되는 걸 수도 있습니다.

**Reply by llIllIllllIIIlllllll** [reply]
주문이 안들어가요...
계속 해보니까 계좌 비밀번호 저장이 안되는거 같아요 
실행하고 비밀번호 보니까 항상 비어있었는데 원래 자동로그인 시키면 저장되어있었는데 계속 풀리는지...
 그것 때문에 주문이 안들어가는지...
잘 모르겠네요...

**Reply by llIllIllllIIIlllllll** [reply]
```python
credentials = {
    'user_id': 'asdf21',
    'user_password': '1234',
    'cert_password': '1234',
    'is_simulation': True,
    'account_passwords': {
        '1234123411': '1234',
    }    
}
entrypoint.EnsureConnected(credentials)
```

이렇게하고 하니까

계좌 비밀번호 입력창을 통해 조회에 사용한 계좌번호의 비밀번호를 입력하십시오 44
이런 오류 뜨네요... 비밀번호 등록이 풀리네요

**Reply by elbakramer** [reply]
1. 일단 주문이 안되는건 말씀주신 것처럼 계좌 비밀번호 설정이 안되어 있어서, 혹은 설정을 했지만 풀렸던게 원인으로 보입니다.
2. 계좌 비밀번호 설정을 하셨더라도 풀리게 되는 원인은, 적어주신 것처럼 `EnsureConnected` 메소드 호출시 인증 정보들을 `dict` 인수로 주면서 호출하는 경우, 해당 인증 정보를 바탕으로 일회성 수동 (매크로) 로그인을 진행하기 전에 기 설정된 자동 로그인 정보를 해제하는 단계가 있는데요. 이것 때문으로 보입니다.
3. 단기 해결책으론 일반적으로 사용하실때는 미리 자동로그인 설정을 해둔 뒤에 `EnsureConnected` 메소드 호출방식을 별도 인자없이 호출하는 방식으로 해서 설정된 자동로그인이 자연스럽게 처리되도록 하시면 좋을 듯 싶습니다. 아니면 `entrypoint.EnsureConnected(credentials)` 호출 이후에 `entrypoint.EnsureAutoLoginEnabled(credentials)` 도 추가로 호출하는 것도 하나의 방법일 듯 싶습니다.
4. 장기적으론 적어주신 것과 같은 `EnsureConnected` 사용방식에서 넘어오는 인자의 인증정보들과 관련해서 계좌 비밀번호도 로그인 이후 함께 처리하도록 `koapy` 쪽 코드를 개선해볼 수 있을 것 같습니다. 이 부분은 제 쪽에서 검토해볼게요.

**Reply by llIllIllllIIIlllllll** [reply]
답변해주셔서 감사합니다
